src/platform/packages/shared/shared-ux/toolbar_selector/src/toolbar_selector.test.tsx

```diff
@@ -0,0 +1,139 @@
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import '@testing-library/jest-dom';
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import type { SelectableEntry } from './toolbar_selector';
+import { ToolbarSelector } from './toolbar_selector';
+
+const options: SelectableEntry[] = [
+  { label: 'Option 1', value: '1' },
+  { label: 'Option 2', value: '2' },
+  { label: 'Option 3', value: '3' },
+];
+
+describe('ToolbarSelector', () => {
+  it('renders button label', () => {
+    const { getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorTest"
+        buttonLabel="Choose"
+        options={options}
+        searchable={false}
+        singleSelection={true}
+      />
+    );
+    expect(getByText('Choose')).toBeInTheDocument();
+  });
+
+  it('opens popover and shows options', () => {
+    const { getByTestId, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorTest"
+        buttonLabel="Select"
+        options={options}
+        searchable={false}
+        singleSelection={true}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorTestButton'));
+    expect(getByText('Option 1')).toBeInTheDocument();
+    expect(getByText('Option 2')).toBeInTheDocument();
+    expect(getByText('Option 3')).toBeInTheDocument();
+  });
+
+  it('calls onChange for single selection', () => {
+    const onChange = jest.fn();
+    const { getByTestId, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorTest"
+        buttonLabel="Select"
+        options={options}
+        searchable={false}
+        singleSelection={true}
+        onChange={onChange}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorTestButton'));
+    fireEvent.click(getByText('Option 2'));
+    expect(onChange).toHaveBeenCalled();
+    expect(onChange.mock.calls[0][0].label).toBe('Option 2');
+  });
+
+  it('calls onChange for multi selection', () => {
+    const onChange = jest.fn();
+    const multiOptions: SelectableEntry[] = [
+      { label: 'A', value: 'a' },
+      { label: 'B', value: 'b' },
+    ];
+    const { getByTestId, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorMultiTest"
+        buttonLabel="Multi"
+        options={multiOptions}
+        searchable={false}
+        singleSelection={false}
+        onChange={onChange}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorMultiTestButton'));
+    fireEvent.click(getByText('A'));
+    fireEvent.click(getByText('B'));
+    expect(onChange).toHaveBeenCalled();
+    // Should be called with array of selected options
+    expect(Array.isArray(onChange.mock.calls[0][0])).toBe(true);
```

Comment:
Could we validate the items that were passed to `onChange`? 

eg:

```ts
 expect(onChange).toHaveBeenCalledWith([
    { label: 'A', value: 'a' },
    { label: 'B', value: 'b' },
  ]);
```

Reply:
Thanks for the suggestions! We have 2 calls, so I used `toHaveBeenNthCalledWith` and checked both calls 👍